src/deluge/io/midi/device_specific/midi_device_lumi_keys.h

```diff
@@ -0,0 +1,194 @@
+ * terms of the GNU General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "io/midi/midi_device.h"
+
+#define MIDI_DEVICE_LUMI_KEYS_VP_COUNT 1
+
+#define MIDI_DEVICE_LUMI_KEYS_SYSEX_START 0xF0
+#define MIDI_DEVICE_LUMI_KEYS_SYSEX_END 0xF7
+#define MIDI_DEVICE_LUMI_KEYS_SYSEX_SPACER 0x77
+#define MIDI_DEVICE_LUMI_KEYS_DEVICE 0x00 // All Devices
+
+#define MIDI_DEVICE_LUMI_KEYS_CONFIG_PREFIX 0x10
+#define MIDI_DEVICE_LUMI_KEYS_MPE_ZONE_PREFIX 0x00
+#define MIDI_DEVICE_LUMI_KEYS_MPE_CHANNELS_PREFIX 0x10
+#define MIDI_DEVICE_LUMI_KEYS_MIDI_MODE_PREFIX 0x20
+#define MIDI_DEVICE_LUMI_KEYS_KEY_PREFIX 0x30
+#define MIDI_DEVICE_LUMI_KEYS_KEY_COUNT 12
+#define MIDI_DEVICE_LUMI_KEYS_SCALE_PREFIX 0x60
+#define MIDI_DEVICE_LUMI_KEYS_SCALE_COUNT 19
+
+#define MIDI_DEVICE_LUMI_KEYS_CONFIG_ROOT_COLOUR_PREFIX 0x30
+#define MIDI_DEVICE_LUMI_KEYS_CONFIG_GLOBAL_COLOUR_PREFIX 0x20
+
+#define MIDI_DEVICE_LUMI_SCALE_MAJOR 0b101010110101
+#define MIDI_DEVICE_LUMI_SCALE_MINOR 0b010110101101
+#define MIDI_DEVICE_LUMI_SCALE_HARMONIC_MINOR 0b100110101101
+#define MIDI_DEVICE_LUMI_SCALE_PENTATONIC_NEUTRAL 0b010010100101
+#define MIDI_DEVICE_LUMI_SCALE_PENTATONIC_MAJOR 0b001010010101
+#define MIDI_DEVICE_LUMI_SCALE_PENTATONIC_MINOR 0b010010101001
+#define MIDI_DEVICE_LUMI_SCALE_BLUES 0b010011101001
+#define MIDI_DEVICE_LUMI_SCALE_DORIAN 0b011010101101
+#define MIDI_DEVICE_LUMI_SCALE_PHRYGIAN 0b010110101011
+#define MIDI_DEVICE_LUMI_SCALE_LYDIAN 0b101011010101
+#define MIDI_DEVICE_LUMI_SCALE_MIXOLYDIAN 0b011010110101
+#define MIDI_DEVICE_LUMI_SCALE_LOCRIAN 0b010101101011
+#define MIDI_DEVICE_LUMI_SCALE_WHOLE_TONE 0b010101010101
+#define MIDI_DEVICE_LUMI_SCALE_ARABIC_A 0b101101101101
+#define MIDI_DEVICE_LUMI_SCALE_ARABIC_B 0b010101110101
+#define MIDI_DEVICE_LUMI_SCALE_JAPANESE 0b100011010001
+#define MIDI_DEVICE_LUMI_SCALE_RYUKYU 0b100010110001
+#define MIDI_DEVICE_LUMI_SCALE_8TONE_SPANISH 0b010101111011
+#define MIDI_DEVICE_LUMI_SCALE_CHROMATIC 0b111111111111
+
+class MIDIDeviceLumiKeys final : public MIDIDeviceUSBHosted {
+public:
+	static constexpr uint16_t lumiKeysVendorProductPairs[MIDI_DEVICE_LUMI_KEYS_VP_COUNT][2] = {{0x2af4, 0xe00}};
+
+	static constexpr uint8_t sysexManufacturer[3] = {0x00, 0x21, 0x10};
+
+	static constexpr uint8_t sysexMidiChannel[16][2] = {
+	    {0x20, 0x00}, // Ch. 1
+	    {0x40, 0x00}, {0x60, 0x00}, {0x00, 0x01}, {0x20, 0x01}, {0x40, 0x01}, {0x60, 0x01}, {0x00, 0x02}, {0x20, 0x02},
```

Comment:
can these be computed instead of hardcoded? it's just channel << 5 then split into 2 bytes for whatever reason

Reply:
Already switched this morning (checked in this evening... was being social today).